internal/info/info_test.go

```diff
@@ -382,352 +556,596 @@ func TestDetail_Get(t *testing.T) {
+			got := i.Get()
+			if err := test.checkFunc(test.want, got); err != nil {
 				tt.Errorf("error = %v", err)
 			}
+
 		})
 	}
 }
 
-func TestInit(t *testing.T) {
-	type args struct {
-		name string
+func Test_info_prepare(t *testing.T) {
+	type fields struct {
+		detail      Detail
+		prepOnce    sync.Once
+		rtCaller    func(skip int) (pc uintptr, file string, line int, ok bool)
+		rtFuncForPC func(pc uintptr) *runtime.Func
 	}
 	type want struct {
-		want *Detail
+		want info
 	}
 	type test struct {
 		name       string
-		args       args
+		fields     fields
 		want       want
-		checkFunc  func(want, *Detail) error
-		beforeFunc func(args)
-		afterFunc  func(args)
+		checkFunc  func(info, want) error
+		beforeFunc func()
+		afterFunc  func()
 	}
-	defaultCheckFunc := func(w want, got *Detail) error {
-		opts := []cmp.Option{
-			cmpopts.IgnoreFields(*w.want, "PrepOnce"),
+	defaultCheckFunc := func(got info, w want) error {
+		opts := []comparator.Option{
+			comparator.AllowUnexported(info{}),
+			comparator.IgnoreFields(info{}, "prepOnce"),
 		}
-		if diff := cmp.Diff(*w.want, *got, opts...); len(diff) != 0 {
+		if diff := comparator.Diff(w.want, got, opts...); len(diff) != 0 {
 			return errors.Errorf("err: %s", diff)
 		}
 		return nil
 	}
 	tests := []test{
 		{
 			name: "set success when all fields are empty",
-			args: args{
-				name: "gateway",
+			want: want{
+				want: info{
+					detail: Detail{
+						GitCommit:  "master",
+						Version:    "gitcommit",
+						BuildTime:  "1s",
+						GoVersion:  runtime.Version(),
+						GoOS:       runtime.GOOS,
+						GoArch:     runtime.GOARCH,
+						CGOEnabled: "true",
+						NGTVersion: "v1.11.6",
+						BuildCPUInfoFlags: []string{
+							"avx512f", "avx512dq",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "GitCommit and Version field is not overwritten when GitCommit field is `internal`",
+			fields: fields{
+				detail: Detail{
+					GitCommit: "internal",
+				},
 			},
 			want: want{
-				want: &Detail{
-					GitCommit:  "gitcommit",
-					ServerName: "gateway",
-					Version:    "gitcommit",
-					BuildTime:  "1s",
-					GoVersion:  runtime.Version(),
-					GoOS:       runtime.GOOS,
-					GoArch:     runtime.GOARCH,
-					CGOEnabled: "true",
-					NGTVersion: "v1.11.6",
-					BuildCPUInfoFlags: []string{
-						"avx512f", "avx512dq",
+				want: info{
+					detail: Detail{
+						GitCommit:  "internal",
+						Version:    "gitcommit",
+						BuildTime:  "1s",
+						GoVersion:  runtime.Version(),
+						GoOS:       runtime.GOOS,
+						GoArch:     runtime.GOARCH,
+						CGOEnabled: "true",
+						NGTVersion: "v1.11.6",
+						BuildCPUInfoFlags: []string{
+							"avx512f", "avx512dq",
+						},
 					},
 				},
 			},
-			beforeFunc: func(args) {
-				GitCommit = "gitcommit"
-				Version = ""
-				BuildTime = "1s"
-				CGOEnabled = "true"
-				NGTVersion = "v1.11.6"
-				BuildCPUInfoFlags = "\t\tavx512f avx512dq\t"
+		},
+
```

Comment:
Could you please add test case of `Version field is not overwritten when GitCommit field is 'v1.0.0'`